apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -254,6 +257,50 @@ public async Task<int> GetOrdinalAsync(string name, CancellationToken cancellati
             return _fieldIndex[name];
         }
 
+        /// <summary>
+        /// Gets the read timestamp if <see cref="TimestampBound.ReturnReadTimestamp" /> was true.
+        /// </summary>
+        /// <remarks>
+        /// The read timestamp can be read before <see cref="SpannerDataReader.Read"/> or
+        /// <see cref="SpannerDataReader.ReadAsync"/> is called.
+        /// </remarks>
+        /// <remarks>
+        /// The read timestamp is available even if no data is returned.
+        /// </remarks>
+        /// <returns>The value converted to a <see cref="Timestamp"/>.</returns>
+        /// <exception cref="InvalidOperationException">A read timestamp is unavailable,
+        /// possibly because <see cref="TimestampBound.ReturnReadTimestamp" /> was false.</exception>
+        public Timestamp GetReadTimestamp()
+        {
+            if (_readTimestamp != null)
+            {
+                return _readTimestamp;
+            }
+            return Task.Run(() => GetReadTimestampAsync(CancellationToken.None)).ResultWithUnwrappedExceptions();
+        }
+
+        /// <summary>
+        /// Gets the read timestamp if <see cref="TimestampBound.ReturnReadTimestamp" /> was true.
+        /// </summary>
+        /// <returns>The value converted to a <see cref="Timestamp"/>.</returns>
+        /// <exception cref="InvalidOperationException">A read timestamp is unavailable,
+        /// possibly because <see cref="TimestampBound.ReturnReadTimestamp" /> was false.</exception>
+        public Task<Timestamp> GetReadTimestampAsync(CancellationToken cancellationToken)
+        {
+            if (_readTimestamp != null)
+            {
+                return Task.FromResult(_readTimestamp);
+            }
+            return ExecuteHelper.WithErrorTranslationAndProfiling(
```

Comment:
Ah ok, I guess that's reasonable. I didn't do it before because this is the single-use transaction case so there was unlikely to be multiple calls to this. But maybe someone will do that and it's better to have it cached.